core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -264,6 +264,11 @@ static void checkClass(Map<String,Path> clazzes, String clazz, Path jarpath) {
                 if (clazz.startsWith("org.apache.xmlbeans")) {
                     return; // https://issues.apache.org/jira/browse/XMLBEANS-499
                 }
+                if (clazz.startsWith("javax.xml.bind") ||
+                    clazz.startsWith("javax.activation") ||
+                    clazz.startsWith("com.sun.activation.registries")) {
+                    return; // java 9 removed from default modules ...
+                }
```

Comment:
I think this is because we must allow now duplicate classes, because the plugins (azure) that need this now bundle JAXB classes as dependency. If you compile with Java 8, you would see duplicate classes (the one in rt.jar and the dependency one). With Java 9 you would only see one.

I am not sure if the whole thing is right to do. If some modules depends on JAXB, the command line to start ES should add {{-addmodules javax.xml.bind}} or similar.